src/darwinMain/kotlin/io/github/oshai/kotlinlogging/DarwinKLogger.kt

```diff
@@ -0,0 +1,43 @@
+package io.github.oshai.kotlinlogging
+
+import kotlinx.cinterop.ptr
+import platform.darwin.OS_LOG_DEFAULT
+import platform.darwin.OS_LOG_TYPE_DEBUG
+import platform.darwin.OS_LOG_TYPE_DEFAULT
+import platform.darwin.OS_LOG_TYPE_ERROR
+import platform.darwin.OS_LOG_TYPE_INFO
+import platform.darwin.__dso_handle
+import platform.darwin._os_log_internal
+import platform.darwin.os_log_t
+import platform.darwin.os_log_type_enabled
+import platform.darwin.os_log_type_t
+
+public class DarwinKLogger(override val name: String, override val underlyingLogger: os_log_t) :
+  KLogger, DelegatingKLogger<os_log_t> {
+
+  override fun at(level: Level, marker: Marker?, block: KLoggingEventBuilder.() -> Unit) {
+    if (isLoggingEnabledFor(level, marker)) {
+      KLoggingEventBuilder().apply(block).run {
+        _os_log_internal(__dso_handle.ptr, OS_LOG_DEFAULT, level.toDarwinLevel(), message)
```

Comment:
```suggestion
        _os_log_internal(__dso_handle.ptr, underlyingLogger, level.toDarwinLevel(), message)
```
I think this might need fixin